release notes/v0.54.0.md

```diff
@@ -0,0 +1,304 @@
+* **Faster Parsing**: `csv.parse` bypasses most of the JavaScript runtime, offering significant speed improvements for large files.
+* **Lower Memory Usage**: both solution support shared memory across virtual users (VUs) with the `fs.open` function.
+* **Flexibility**: choose between full-file parsing with csv.parse() or memory-efficient streaming with csv.Parser.
+
+#### Tradeoffs
+
+* **`csv.Parse`**: Parses the entire file in the initialization phase of the test, which can increase startup time and memory usage for large files. Best suited for scenarios where performance is prioritized over memory consumption.
+* **`csv.Parser`**: Reads the file line-by-line, making it more memory-efficient but potentially slower due to reading overhead for each line. Ideal for scenarios where memory usage is a concern or where fine-grained control over parsing is needed.
+ 
+#### Example Usage
+
+<details>
+<summary> Expand to see an example of Parsing a full CSV file into a SharedArray.</summary>
+
+```javascript
+import { open } from 'k6/experimental/fs'
+import csv from 'k6/experimental/csv'
+import { scenario } from 'k6/execution'
+
+export const options = {
+    iterations: 10,
+}
+
+let file;
+let csvRecords;
+(async function () {
+    file = await open('data.csv');
+
+    // The `csv.parse` function consumes the entire file at once, and returns
+    // the parsed records as a SharedArray object.
+    csvRecords = await csv.parse(file, {delimiter: ','})
+})();
+
+
+export default async function() {
+    // The csvRecords a SharedArray. Each element is a record from the CSV file, represented as an array
+    // where each element is a field from the CSV record.
+    //
+    // Thus, `csvRecords[scenario.iterationInTest]` will give us the record for the current iteration.
+    console.log(csvRecords[scenario.iterationInTest])
+}
+```
+
+</details>
+
+<details>
+<summary> Expand to see an example of streaming a CSV file line-by-line.</summary>
+
+```javascript
+import { open } from 'k6/experimental/fs'
+import csv from 'k6/experimental/csv'
+
+export const options = {
+    iterations: 10,
+}
+
+let file;
+let parser;
+(async function () {
+    file = await open('data.csv');
+    parser = new csv.Parser(file);
+})();
+
+export default async function() {
+    // The parser `next` method attempts to read the next row from the CSV file.
+    //
+    // It returns an iterator-like object with a `done` property that indicates whether
+    // there are more rows to read, and a `value` property that contains the row fields
+    // as an array.
+    const {done, value} = await parser.next();
+    if (done) {
+        throw new Error("No more rows to read");
+    }
+
+    // We expect the `value` property to be an array of strings, where each string is a field
+    // from the CSV record.
+    console.log(done, value);
+}
+```
+
+</details>
+
+### New `k6 cloud upload` command for uploading test files to the cloud [`#3906`](https://github.com/grafana/k6/pull/3906)
```

Comment:
I think we still need a docs PR for this update.

Reply:
@joanlopez is away, I'll take care of it as soon as possible 👍🏻

Reply:
I've opened https://github.com/grafana/k6-docs/pull/1734 accordingly 🙇🏻